### DISCUSSION TITLE
Equivalent of useLocation and useSearchParams for react

### DISCUSSION THREAD

**Discussion by HSchmale16** [vikejs/vike#1083] [Discussion]
### Description

What's the easiest way to set query string values within the server side rendering framework?

I got build errors with react router, but I need certain pages to be prerendered for seo purposes.

Specifically I'm referring to this page: https://vite-plugin-ssr.com/route-string

I can't find anything about working with query strings and locations here.

**Comment by brillout** [comment]
I'm not sure I understand your question. Do you mean that want to define a parameterized Route String using query params such as `/product?@id` instead of `/product/@id`?

**Reply by brillout** [reply]
As for accessing the query params, see https://vite-plugin-ssr.com/pageContext and https://vite-plugin-ssr.com/pageContext-anywhere.
